packages/sanity/src/desk/panes/document/documentPanel/header/DocumentHeaderTitle.tsx

```diff
@@ -1,32 +1,18 @@
 import React, {ReactElement} from 'react'
+import {useDocumentTitle} from '../../useDocumentTitle'
 import {useDocumentPane} from '../../useDocumentPane'
-import {unstable_useValuePreview as useValuePreview} from 'sanity'
 
 export function DocumentHeaderTitle(): ReactElement {
-  const {connectionState, schemaType, title, value: documentValue} = useDocumentPane()
-  const subscribed = Boolean(documentValue) && connectionState === 'connected'
-
-  const {error, value} = useValuePreview({
-    enabled: subscribed,
-    schemaType,
-    value: documentValue,
-  })
+  const {connectionState} = useDocumentPane()
+  const {error, title} = useDocumentTitle()
 
   if (connectionState !== 'connected') {
     return <></>
   }
 
-  if (title) {
-    return <>{title}</>
-  }
-
-  if (!documentValue) {
-    return <>New {schemaType?.title || schemaType?.name}</>
-  }
-
   if (error) {
-    return <>Error: {error.message}</>
+    return <>{error}</>
   }
 
-  return <>{value?.title || <span style={{color: 'var(--card-muted-fg-color)'}}>Untitled</span>}</>
+  return <>{title || <span style={{color: 'var(--card-muted-fg-color)'}}>Untitled</span>}</>
```

Comment:
Small nitpick: Can we move the style out of the inline document?

Reply:
actually, after looking at this, I realized that I will be refactoring this in https://github.com/sanity-io/sanity/pull/4689 so it might not be worth changing it in this PR